File: cpp/1726. 同积元组.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& nums) {
        unordered_map<int, int> mp;
        int n=nums.size();
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                mp[nums[i]*nums[j]]++;
            }
        }
        int ans=0;
        for(auto &[k, v]: mp){
            ans+=(v-1)*v*4;
        }
        return ans;
    }
};

File: 刷题/golang/1726. 同积元组.go
func tupleSameProduct(nums []int) int {
    mp := make(map[int]int)
    n := len(nums)
    for i:=0; i<n; i++ {
        for j:=i+1; j<n; j++ {
            mp[nums[i]*nums[j]]++
        }
    }
    ans := 0
    for _, v := range mp {
        ans += v*(v-1) * 4
    }
    return ans
}